zypp/zyppng/base/private/threaddata_p.h

```diff
@@ -4,38 +4,49 @@
 #include <memory>
 #include <thread>
 
-namespace zyppng {
-
+namespace zyppng
+{
   class EventDispatcher;
 
-  struct ThreadData {
-    static ThreadData &current ();
+  struct ThreadData
+  {
+    static ThreadData &current();
 
     template<typename T>
-    void setName ( T &&name ) {
-      threadName = std::forward<T>( name );
+    void setName( T &&name ) {
+      _threadName = std::forward<T>( name );
       syncNativeName();
     }
 
-    std::string name () const;
+    const std::string &name() const;
+
+    /** Set hint for logger to hide the thread name. */
+    void setHideNameHint( bool yesno )
+    { _hideNameHint = yesno; }
```

Comment:
Can we do that in LogControl? That is not really something that should be in ThreadData. I'd prefer we do not use that as location for random settings. Each Thread has its own LogControlImpl instance, so settings for the logger should be there

Reply:
Good point.